ui/src/components/mobileservices/validators/AbstractBindingValidator.js

```diff
@@ -0,0 +1,68 @@
+/* eslint no-continue: off */
+/* eslint class-methods-use-this: off */
+export class AbstractBindingValidator {
+  constructor(platform, customProperties = {}) {
+    this.customProperties = customProperties;
+
+    this.detectPlatform = customProperties.platformDetector;
+    this.errorsKey = customProperties.errorField;
+    this.platform = platform;
+  }
+
+  addError(errors, key, message) {
+    errors[this.errorsKey || key].addError(message);
+  }
+
+  preFieldValidation(key, value) {
+    if (this.customProperties.preValidate) {
+      return this.customProperties.preValidate(key, value);
+    }
+    return false;
+  }
+
+  validateField(key, value) {
+    return false;
+  }
+
+  postFieldValidation(key, value) {
+    if (this.customProperties.postValidate) {
+      return this.customProperties.postValidate(key, value);
+    }
+    return false;
+  }
+
+  validate(formData, errors) {
+    let hasErrors = false;
+    if (this.platform && this.detectPlatform && this.detectPlatform() !== this.platform) {
+      return false;
+    }
+
+    for (const key in errors) {
+      if (errors.hasOwnProperty(key)) {
+        const value = formData[key];
+
+        let error = this.preFieldValidation(key, value);
+        if (error) {
+          hasErrors = true;
+          this.addError(errors, key, error);
+          continue;
+        }
+
+        error = this.validateField(key, value);
+        if (error) {
+          hasErrors = true;
+          this.addError(errors, key, error);
+          continue;
+        }
+
+        error = this.postFieldValidation(key, value);
```

Comment:
given that all the validators are called in a single loop, and the input and return values are exactly the same, I am not sure if there is much value to separate the validation methods to 3 different ones. It will also cause confusion on when to use which?

Reply:
The `preFieldValidation` and `postFieldValidation` are custom function that can be passed to the validators to perform custom `pre` or `post` validations.

Reply:
yes, but why custom? WE already have the concrete class to validate for android and ios, why split the validation logic in 2 different places?

Reply:
To keep the interface consistent even in case we will have to add other validators.
Currently all the validations requires only the `formData` object, while only the password confirmation requires the `formData`, `form` and the `document` (see [here](https://github.com/ziccardi/mobile-developer-console/blob/caabe8e32372a6166a770e04575aaf6e58a92d5d/ui/src/components/mobileservices/BindingPanel.js#L146-L147) ). This way we can have a coherent interface even for future validators requiring other custom validations.